### PR TITLE
Migrate to Gradle

### PR DESCRIPTION
**Pull request by Egorand**

### CODE REVIEW

**Comment by Egorand** [comment]
@cwwwu can you please 👀 ? Can figure out snapshot credentials later

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+jar {
+  manifest {
+    attributes('Automatic-Module-Name': 'com.squareup.kotlinpoet')
+  }
+}
+
+dokka {
+  skipDeprecated = true
```
Comment (hzsweers) [inline build.gradle]: TIL

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+apply plugin: 'org.jetbrains.dokka'
+apply plugin: 'com.vanniktech.maven.publish'
+
+repositories {
+  mavenCentral()
+}
+
+jar {
+  manifest {
+    attributes('Automatic-Module-Name': 'com.squareup.kotlinpoet')
```
Comment (hzsweers) [inline build.gradle]: out of curiosity - what was this for?

Comment (Egorand) [reply build.gradle]: http://branchandbound.net/blog/java/2017/12/automatic-module-name/

TLDR: It's needed for Java 9's module system (that somebody somewhere somehow probably uses)
